ui/src/main/java/org/pentaho/di/ui/trans/steps/pentahoreporting/PentahoReportingOutputDialog.java

```diff
@@ -154,34 +163,96 @@ public void widgetSelected( SelectionEvent e ) {
     fdStepname.right = new FormAttachment( 100, 0 );
     wStepname.setLayoutData( fdStepname );
 
+    // input file field line (report definition)
+    wlInputField = new Label( shell, SWT.RIGHT );
+    wlInputField.setText( BaseMessages.getString( PKG, "PentahoReportingOutputDialog.InputFilenameField.Label" ) );
+    props.setLook( wlInputField );
+    FormData fdlInputField = new FormData();
+    fdlInputField.left = new FormAttachment( 0, 0 );
+    fdlInputField.top = new FormAttachment( wStepname, margin + 5 );
+    fdlInputField.right = new FormAttachment( middle, -margin );
+    wlInputField.setLayoutData( fdlInputField );
+    wInputField = new Combo( shell, SWT.SINGLE | SWT.LEFT | SWT.BORDER );
+    props.setLook( wInputField );
+    wInputField.addModifyListener( lsMod );
+    FormData fdInputField = new FormData();
+    fdInputField.left = new FormAttachment( middle, 0 );
+    fdInputField.top = new FormAttachment( wStepname, margin + 5 );
+    fdInputField.right = new FormAttachment( 100, 0 );
+    wInputField.setLayoutData( fdInputField );
+
+    String[] fieldNames = new String[] {};
+    try {
+      fieldNames = Const.sortStrings( transMeta.getPrevStepFields( stepMeta ).getFieldNames() );
+    } catch ( KettleException e ) {
+      log.logError( "Unexpected error getting fields from previous steps...", e );
```

Comment:
Use BaseMessages.getString for this string. put it in a .properties file

Reply:
done